cpp/src/table/row_operators.cu

```diff
@@ -175,6 +270,28 @@ void check_lex_compatibility(table_view const& input)
   }
 }
 
+/**
+ * @brief Check a table for compatibility with equality comparison
+ *
+ * Checks whether a given table contains columns of non-equality comparable types.
+ */
+void check_eq_compatibility(table_view const& input)
+{
+  std::function<void(column_view const&)> check_column = [&](column_view const& c) {
```

Comment:
Maybe worth a `using` declaration of this function type to reuse with above, especially if you're going to need another `check_hash_compatibility` eventually?